nsiqcppstyle_rulemanager.py

```diff
@@ -180,62 +200,154 @@ def ResetRegisteredRules(self):
         del self.typeScopeRules[:]
         del self.fileStartRules[:]
         del self.fileEndRules[:]
+        del self.sessionStartRules[:]

```

Comment:
We should use `self.rules.clear()` instead of `del`

Reply:
Done.